draft-yasskin-httpbis-origin-signed-exchanges-impl.md

```diff
@@ -902,40 +949,43 @@ Cross origin push is not implemented.
 
 To allow signed exchanges to be the targets of `<link rel=prefetch>` tags, we
 define the  `application/signed-exchange` content type that represents a signed
-HTTP exchange, including request metadata and header fields, response metadata
+HTTP exchange, including a request URL, response metadata
 and header fields, and a response payload.
 
+When served over HTTP, a response containing an `application/signed-exchange`
+payload MUST include at least the following response header fields, to reduce
+content sniffing vulnerabilities:
+
+* Content-Type: application/signed-exchange;v=*version*
+* X-Content-Type-Options: nosniff
+
 This content type consists of the concatenation of the following items:
 
-1. The ASCII characters "sxg1-b2" followed by a 0 byte, to serve as a file
-   signature. This is redundant with the MIME type, and recipients that receive
-   both MUST check that they match and stop parsing if they don't.
+1. 8 bytes consisting of the ASCII characters "sxg1-b3" followed by a 0 byte,
+   to serve as a file signature. This is redundant with the MIME type, and
+   recipients that receive both MUST check that they match and stop parsing if
+   they don't.
```

Comment:
Does this imply that implementation must not fallback-redirect if they don't match?

Reply:
Current Chromium implementation does fallback redirect in this case.

Let's continue discussing in #397 on when we should fallback.